crates/ursa-store/src/store.rs

```diff
@@ -129,11 +133,16 @@ where
     }
 }
 
+#[async_trait]
 pub trait Dag {
     /// traverse a dag and get full dag given a root cid
     fn dag_traversal(&self, root_cid: &Cid) -> Result<Vec<(Cid, Vec<u8>)>>;
+
+    /// Build a temporary car file from a root cid and return the size
+    async fn car_size(&self, root_cid: &Cid) -> Result<u64>;
```

Comment:
This trait seemed to provide a generic dag interface but this method seems specific to car files. Maybe it would fit better as a method of `UrsaStore`.

Reply:
Hmm, I had initially thought it fit there since it directly uses traverse_dag to build the file, I can move it to ursastore directly but just double checking

Reply:
And now that I'm thinking about it. The only implementer of `Dag` is `UrsaStore` so I don't think we need abstract `dag_traversal` and `car_size`. I recommend that we simplify this and move these methods in `UrsaStore`.

Reply:
done. I actually put it on BlockstoreExt for a more generic addition https://github.com/fleek-network/ursa/pull/278/commits/04b7d8f83c37e68f7c29250dc04f4e4d7e53793a